### PR TITLE
FileDistributor now works with files that have to be in subdirectories.

### PR DESCRIPTION
**Pull request by mhrivnak**
https://pulp.plan.io/issues/1824

fixes #1824

### CODE REVIEW

File: server/test/unit/plugins/file/test_distributor.py [hunk 1]
```diff
@@ -216,6 +216,55 @@ def test__symlink_units(self):
...
+        mock_makedirs.assert_called_once_with('/a/b/c/d/e')
+
+    @patch('os.path.exists', return_value=False)
+    @patch('os.makedirs')
+    @patch('os.symlink')
+    def test__symlink_unit_with_existing_subdir(self, mock_symlink, mock_makedirs, mock_exists):
+        """
+        Make sure that if the file being published has to go in a subdirectory, that subdirectory
+        gets created.
+        """
```
Comment (goosemania) [inline server/test/unit/plugins/file/test_distributor.py]: this docstring probably came from the previous test... subdirectory won't be created in this test

File: server/test/unit/plugins/file/test_distributor.py [hunk 2]
```diff
@@ -216,6 +216,55 @@ def test__symlink_units(self):
...
+        mock_makedirs.assert_called_once_with('/a/b/c/d/e')
+
+    @patch('os.path.exists', return_value=False)
+    @patch('os.makedirs')
+    @patch('os.symlink')
+    def test__symlink_unit_with_subdir_exception(self, mock_symlink, mock_makedirs, mock_exists):
+        """
+        Make sure that if the file being published has to go in a subdirectory, that subdirectory
+        gets created.
+        """
```
Comment (goosemania) [inline server/test/unit/plugins/file/test_distributor.py]: Same here, wrong docstring

**Comment by mhrivnak** [comment]
ok test

**Comment by mhrivnak** [comment]
ok test

**Comment by mhrivnak** [comment]
It looks like none of the test failures are my fault, so I'm going to merge.

**Pull request merged**